hudi-common/src/main/java/org/apache/hudi/common/table/log/HoodieMergedLogRecordScanner.java

```diff
@@ -95,6 +99,7 @@ protected HoodieMergedLogRecordScanner(FileSystem fs, String basePath, List<Stri
       this.records = new ExternalSpillableMap<>(maxMemorySizeInBytes, spillableMapBasePath, new DefaultSizeEstimator(),
           new HoodieRecordSizeEstimator(readerSchema), diskMapType, isBitCaskDiskMapCompressionEnabled);
       this.maxMemorySizeInBytes = maxMemorySizeInBytes;
+      this.hoodieMerge = HoodieRecordUtils.loadHoodieMerge(getMergeClassFQN());
```

Comment:
when writing from Spark we would use `HoodieSparkRecord` and the merge class would be `HoodieSparkMerge` in the table properties? and if we now read with Hive, it falls back to avro somehow? 

Reply:
Yeah，some config in the table properties will choose the record type and merge class.

Reply:
In Step3, the reader/writer will choose the corresponding record type if the user does not specify it directly.